lib/pacemaker/pcmk_output.c

```diff
@@ -2141,6 +2141,13 @@ attribute_default(pcmk__output_t *out, va_list args)
 
     GString *s = g_string_sized_new(50);
 
+    char *value_esc = NULL;
+
+    if (pcmk__xml_needs_escape(value, true)) {
+        value_esc = pcmk__xml_escape(value, true);
+        value = value_esc;
+    }
```

Comment:
Set a multi-line attribute:
```
$ sudo attrd_updater -N laptop -n openstack_ports -U "$(echo -e "subnet_id_a\nsubnet_id_b:port_id_1,subnet_id_c\nsubnet_id_d:port_id_2")"
```

Query before:
```
$ sudo attrd_updater -N laptop -n openstack_ports -Q
name="openstack_ports" host="laptop" value="subnet_id_a
subnet_id_b:port_id_1,subnet_id_c
subnet_id_d:port_id_2"
```

Query after:
```
$ sudo attrd_updater -N laptop -n openstack_ports -Q 
name="openstack_ports" host="laptop" value="subnet_id_a&#x0A;subnet_id_b:port_id_1,subnet_id_c&#x0A;subnet_id_d:port_id_2"
```

Note that this change is conservative: it applies only to outputting attributes in text format. Also, dumping XML to text (i.e., to a string representation of the XML) already escaped special characters as appropriate throughout the document.

Any other path through which an attribute value might be represented as text, will not escape newlines (etc.) if it didn't escape them before.

Reply:
BTW, the values may be fetched from `attrd` rather than from the CIB XML, which is why they're not escaped when we get them. (Another possibility would be to escape them when we store them in attrd, but I'm inclined to keep the characters the way they were received unless escapes are needed for XML or display.)